### PR TITLE
Fix for common height issues in Drawer component.

### PR DESCRIPTION
**Pull request by sambev**
- Set a static height (50px) to the header
- Use calc to assign dynamic height for the content
- set overflow on the content.

### CODE REVIEW

**Comment by jmophoto** [comment]
What about making `header_height` a prop with a default value of 50? Then it could be easily overridden if needed.

**Comment by sambev** [comment]
Good point. That would be more extensible than what I have, but the API feels a bit strange since one can already set the height via the prop `headerStyle`.

Which brings up another problem with my approach, it can get nuked by someone passing in `height` via `headerStyle` and the content wont account for it.

https://github.com/mxenabled/mx-react-components/blob/ba7d39e94ef10b2b14556f7245a8de168338826a/src/components/Drawer.js#L171

 Maybe I could just set a default of 50 to `height` in `headerStyle`? That way we don't have two props that can clash and the body content will pick it up?

File: src/components/Drawer.js [hunk 1]
```diff
@@ -229,7 +231,8 @@ class Drawer extends React.Component {
       },
       content: {
         backgroundColor: StyleConstants.Colors.WHITE,
-        height: '100%'
+        overflow: 'auto',
+        height: `calc(100% - ${HEADER_HEIGHT})`
```
Comment (cerinman) [inline src/components/Drawer.js]: I'm not sure how well supported `calc` is for mobile devices.  There is `?` for android, IE, and Opera.

https://developer.mozilla.org/en-US/docs/Web/CSS/calc

Comment (cerinman) [inline src/components/Drawer.js]: Also, Firefox and Safari require prefixes it looks like.

Comment (sambev) [inline src/components/Drawer.js]: I texted this on safari mobile, so I know it works there. According to https://caniuse.com/#search=calc It seems that this should work on everything mobile outside of opera, which I am okay with 😬 .

Comment (cerinman) [inline src/components/Drawer.js]: Cool, the extra testing makes me feel better.  Thanks for digging.

Comment (sambev) [inline src/components/Drawer.js]: Tested on 4.3 android, firefox, and safari and they all worked for me.

File: src/components/Drawer.js [hunk 2]
```diff
@@ -96,6 +97,21 @@ class Drawer extends React.Component {
...
 
   /**
+   * Figure out the height of the header. This can come from either:
+   *
+   * this.props.headerStyle.height
+   * this.props.header.height
+   *
+   * If neither are set, default to 50px
+   */
+  _getHeaderHeight = () => {
```
Comment (sambev) [inline src/components/Drawer.js]: @jmophoto I moved the height setting to this method. Let me know what you think

**Comment by cerinman** [comment]
:shipit:

**Pull request merged**